File: inflate/tests/langs/pascal/box.pas
unit Box;

interface

implementation

function helper(value: Integer): Integer;
begin
  Result := value + 1;
end;

function greet(value: Integer): Integer;
begin
  Result := helper(value) * 2;
end;

function runner: Integer;
begin
  Result := greet(3);
end;

end.

File: inflate/tests/langs/fortran/box.f90
module Box
contains

   function helper(value) result(r)
      integer :: value, r
      r = value + 1
   end function

   function greet(value) result(r)
      integer :: value, r
      r = helper(value) * 2
   end function

   function runner() result(r)
      integer :: r
      r = greet(3)
   end function

end module Box
